tests/Functional/Cases/Rendering/DataAccessorTest.php

```diff
@@ -0,0 +1,207 @@
+<?php
+namespace TYPO3Fluid\Fluid\Tests\Functional\Cases\Rendering;
+
+use TYPO3Fluid\Fluid\Core\Cache\FluidCacheInterface;
+use TYPO3Fluid\Fluid\Core\Cache\SimpleFileCache;
+use TYPO3Fluid\Fluid\Tests\Functional\Cases\Rendering\Fixtures as Fixtures;
+use TYPO3Fluid\Fluid\Tests\UnitTestCase;
+use TYPO3Fluid\Fluid\View\TemplateView;
+
+class DataAccessorTest extends UnitTestCase
+{
+    /**
+     * @return array
+     */
+    public function dataIsRenderedDataProvider()
+    {
+        return [
+            // array values
+            [
+                ['value' => 'value'],
+                ['value' => 'value'],
+            ],
+            [
+                new \ArrayObject(['value' => 'value']),
+                ['value' => 'value'],
+            ],
+            // accessing using properties directly
+            [
+                $this->createObjectWithProperties(),
+                [
+                    'privateValue' => null,
+                    'protectedValue' => null,
+                    'publicValue' => 'publicValue',
+                ],
+                'Cannot access private property class@anonymous::$privateValue', // @todo Probably regression?!
```

Comment:
To my knowledge, not a regression. Attempting to access private or protected properties without getter methods has always triggered a PHP error (we just never tested for it).